cmake/ADIOSFunctions.cmake

```diff
@@ -22,11 +22,20 @@ function(message_pad msg out_len out_msg)
   endif()
 endfunction()
 
-function(python_add_test name script)
-  add_test(NAME ${name}
-    COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/${script} ${ARGN}
+function(python_add_test)
+  set(options)
+  set(oneValueArgs
+      TARGET
   )
-  set_property(TEST ${name} PROPERTY
+  set(multiValueArgs
+      EXEC_WRAPPER
+      SCRIPT
+  )
```

Comment:
Since there's only a few of these they can easily fit on a single line.  The multi-line breakout is really to deal with long lines > 80c.

```cmake
set(options)
set(oneValueArgs TARGET)
set(multiValueArgs EXEC_WRAPPER SCRIPT)
```

That being said, is `SCRIPT` actually a multi-value argument?  What would happen if you passed several scripts here?  it seems things would break on multiple levels.  This would likely be better suited to dropping the `TARGET` option entirely and just having the multi-value `EXEC_WRAPPER` and `SCRIPTS` arguments.  Then loop through and add a test for each one by extracting the test name from filename by dropping the extension (see [get_filename_component](https://cmake.org/cmake/help/v3.6/command/get_filename_component.html)).